// ng2-progress-bar/index.js
"use strict";
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var ProgressBar = (function () {
    function ProgressBar() {
        this.value = 0;
        this.max = 100;
    }
    ProgressBar.prototype.getProgress = function () {
        if (this.max === 0)
            return 0;
        return this.value / this.max * 100;
    };
    ProgressBar.decorators = [
        { type: core_1.Component, args: [{
                    selector: "progress-bar",
                    template: "\n<div class=\"progress\">\n    <div class=\"progress-bar\"\n        role=\"progressbar\"\n        [attr.aria-valuenow]=\"getProgress()\"\n        aria-valuemin=\"0\"\n        [attr.aria-valuemax]=\"getProgress()\" [ngStyle]=\"{ width: getProgress() + '%' }\">\n        <span [hidden]=\"!title\" class=\"progress-bar-title\">{{ title }}</span>\n        <span class=\"sr-only\">{{ getProgress() }}% Complete</span>\n    </div>\n</div>\n"
                },] },
    ];
    /** @nocollapse */
    ProgressBar.ctorParameters = function () { return []; };
    ProgressBar.propDecorators = {
        'value': [{ type: core_1.Input },],
        'max': [{ type: core_1.Input },],
        'title': [{ type: core_1.Input },],
    };
    return ProgressBar;
}());
exports.ProgressBar = ProgressBar;
var ProgressBarModule = (function () {
    function ProgressBarModule() {
    }
    ProgressBarModule.decorators = [
        { type: core_1.NgModule, args: [{
                    imports: [
                        common_1.CommonModule
                    ],
                    declarations: [
                        ProgressBar
                    ],
                    exports: [
                        ProgressBar
                    ]
                },] },
    ];
    /** @nocollapse */
    ProgressBarModule.ctorParameters = function () { return []; };
    return ProgressBarModule;
}());
exports.ProgressBarModule = ProgressBarModule;
